src/brsTypes/components/RoSGNode.ts

```diff
@@ -1914,3 +1925,55 @@ function addChildren(
         }
     });
 }
+
+function cloneNode(
+    interpreter: Interpreter,
+    toBeCloned: RoSGNode,
+    isDeepCopy: Boolean
+): RoSGNode | BrsInvalid {
+    let copy = createNodeByType(interpreter, new BrsString(toBeCloned.nodeSubtype));
+
+    let originalFields = toBeCloned.getFields();
+    let copiedFields = new RoAssociativeArray([]);
+    let addReplace = copiedFields.getMethod("addreplace");
+
+    // Copy the fields.
+    for (let [key, value] of originalFields) {
+        if (addReplace) {
+            addReplace.call(
+                interpreter,
+                new BrsString(key),
+                getBrsValueFromFieldType(value.getType(), value.toString())
+            );
+        }
+    }
+
+    if (!(copy instanceof BrsInvalid)) {
+        let update = copy.getMethod("update");
+
+        if (update) {
+            update.call(interpreter, copiedFields, BrsBoolean.True);
+        }
+
+        // A deep clone also copies children.
+        if (isDeepCopy) {
+            let appendChild = copy.getMethod("appendchild");
+            let getChildren = toBeCloned.getMethod("getchildren");
```

Comment:
This code would be a little cleaner if we exposed `getChildren` in typescript. I wonder if that's worth it. Then the callable would just wrap that response in an `RoArray`.